ModelSnapshot: Ensure using directives present for annotation types

I just installed the beta7-13841 and ran the `Script-Migration' command. It was working with the previous release, now I receive the following error: 

PM> Script-Migration
Specified cast is not valid.

Update: same behavior with Add-Migration

Same here on beta7-12274
```
E:\Projects\HelloMvc\Migrations\PersonDBContextModelSnapshot.cs(16,66): DNX,Version=v4.5.1 error CS0103: The name 'SqlServerIdentityStrategy' does not exist in the current context
E:\Projects\HelloMvc\Migrations\PersonDBContextModelSnapshot.cs(22,74): DNX,Version=v4.5.1 error CS0103: The name 'SqlServerIdentityStrategy' does not exist in the current context
E:\Projects\HelloMvc\Migrations\20150726005437_MyFirstMigration.cs(17,74): DNX,Version=v4.5.1 error CS0103: The name 'SqlServerIdentityStrategy' doesnot exist in the current context
E:\Projects\HelloMvc\Migrations\20150726005437_MyFirstMigration.Designer.cs(26,66): DNX,Version=v4.5.1 error CS0103: The name 'SqlServerIdentityStrategy' does not exist in the current context
E:\Projects\HelloMvc\Migrations\20150726005437_MyFirstMigration.Designer.cs(32,74): DNX,Version=v4.5.1 error CS0103: The name 'SqlServerIdentityStrategy' does not exist in the current context
```

With 7.0.0-beta7-13854 I cleared out all migrations and ran the following updated command:
```
dnx . ef migrations add MyFirstMigration --context PersonDBContext
```

I received these errors when compiling:
```
E:\Projects\HelloMvc\Migrations\PersonDBContextModelSnapshot.cs(16,66): DNX,Version=v4.5.1 error CS0103: The name 'SqlServerIdentityStrategy' does not exist in the current context
E:\Projects\HelloMvc\Migrations\PersonDBContextModelSnapshot.cs(22,74): DNX,Version=v4.5.1 error CS0103: The name 'SqlServerIdentityStrategy' does not exist in the current context
E:\Projects\HelloMvc\Migrations\20150730033506_MyFirstMigration.cs(17,74): DNX,Version=v4.5.1 error CS0103: The name 'SqlServerIdentityStrategy' doesnot exist in the current context
E:\Projects\HelloMvc\Migrations\20150730033506_MyFirstMigration.Designer.cs(26,66): DNX,Version=v4.5.1 error CS0103: The name 'SqlServerIdentityStrategy' does not exist in the current context
E:\Projects\HelloMvc\Migrations\20150730033506_MyFirstMigration.Designer.cs(32,74): DNX,Version=v4.5.1 error CS0103: The name 'SqlServerIdentityStrategy' does not exist in the current context
```
doing a little digging I found 'SqlServerIdentityStrategy' in Microsoft.Data.Entity.SqlServer.Metadata so I added this to all three migration files:
```
using Microsoft.Data.Entity.SqlServer.Metadata;
```
Now when I compile I am now get this error:
```
E:\Projects\HelloMvc\Migrations\20150730033506_MyFirstMigration.cs(18,74): DNX,Version=v4.5.1 error CS1503: Argument 2: cannot convert from 'Microsoft.Data.Entity.SqlServer.Metadata.SqlServerIdentityStrategy' to 'string'
```

Noral

--- a/src/EntityFramework.Commands/Migrations/CSharpMigrationGenerator.cs
+++ b/src/EntityFramework.Commands/Migrations/CSharpMigrationGenerator.cs
@@ -3,6 +3,7 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using JetBrains.Annotations;
 using Microsoft.Data.Entity.Commands.Utilities;
 using Microsoft.Data.Entity.Internal;
@@ -46,11 +47,23 @@ namespace Microsoft.Data.Entity.Commands.Migrations
             Check.NotNull(downOperations, nameof(downOperations));
 
             var builder = new IndentedStringBuilder();
+            var namespaces = new List<string>
+            {
+                "System",
+                "System.Collections.Generic",
+                "Microsoft.Data.Entity.Migrations",
+                "Microsoft.Data.Entity.Migrations.Builders",
+                "Microsoft.Data.Entity.Migrations.Operations"
+            };
+            namespaces.AddRange(GetNamespaces(upOperations.Concat(downOperations)));
+            foreach (var n in namespaces.Distinct())
+            {
+                builder
+                    .Append("using ")
+                    .Append(n)
+                    .AppendLine(";");
+            }
             builder
-                .AppendLine("using System.Collections.Generic;")
-                .AppendLine("using Microsoft.Data.Entity.Migrations;")
-                .AppendLine("using Microsoft.Data.Entity.Migrations.Builders;")
-                .AppendLine("using Microsoft.Data.Entity.Migrations.Operations;")
                 .AppendLine()
                 .Append("namespace ").AppendLine(_code.Namespace(migrationNamespace))
                 .AppendLine("{");
@@ -102,12 +115,23 @@ namespace Microsoft.Data.Entity.Commands.Migrations
             Check.NotNull(targetModel, nameof(targetModel));
 
             var builder = new IndentedStringBuilder();
+            var namespaces = new List<string>
+            {
+                "System",
+                "Microsoft.Data.Entity",
+                "Microsoft.Data.Entity.Metadata",
+                "Microsoft.Data.Entity.Migrations.Infrastructure",
+                contextType.Namespace
+            };
+            namespaces.AddRange(GetNamespaces(targetModel));
+            foreach (var n in namespaces.Distinct())
+            {
+                builder
+                    .Append("using ")
+                    .Append(n)
+                    .AppendLine(";");
+            }
             builder
-                .AppendLine("using System;")
-                .AppendLine("using Microsoft.Data.Entity;")
-                .AppendLine("using Microsoft.Data.Entity.Metadata;")
-                .AppendLine("using Microsoft.Data.Entity.Migrations.Infrastructure;")
-                .Append("using ").Append(contextType.Namespace).AppendLine(";")
                 .AppendLine()
                 .Append("namespace ").AppendLine(_code.Namespace(migrationNamespace))
                 .AppendLine("{");
@@ -166,12 +190,23 @@ namespace Microsoft.Data.Entity.Commands.Migrations
             Check.NotNull(model, nameof(model));
 
             var builder = new IndentedStringBuilder();
+            var namespaces = new List<string>
+            {
+                "System",
+                "Microsoft.Data.Entity",
+                "Microsoft.Data.Entity.Metadata",
+                "Microsoft.Data.Entity.Migrations.Infrastructure",
+                contextType.Namespace
+            };
+            namespaces.AddRange(GetNamespaces(model));
+            foreach (var n in namespaces.Distinct())
+            {
+                builder
+                    .Append("using ")
+                    .Append(n)
+                    .AppendLine(";");
+            }
             builder
-                .AppendLine("using System;")
-                .AppendLine("using Microsoft.Data.Entity;")
-                .AppendLine("using Microsoft.Data.Entity.Metadata;")
-                .AppendLine("using Microsoft.Data.Entity.Migrations.Infrastructure;")
-                .Append("using ").Append(contextType.Namespace).AppendLine(";")
                 .AppendLine()
                 .Append("namespace ").AppendLine(_code.Namespace(modelSnapshotNamespace))
                 .AppendLine("{");

--- a/src/EntityFramework.Commands/Migrations/CSharpModelGenerator.cs
+++ b/src/EntityFramework.Commands/Migrations/CSharpModelGenerator.cs
@@ -17,12 +17,6 @@ namespace Microsoft.Data.Entity.Commands.Migrations
     {
         private readonly CSharpHelper _code;
 
-        private static readonly List<string> _ignoredAnnotations = new List<string>
-        {
-            CoreAnnotationNames.OriginalValueIndexAnnotation,
-            CoreAnnotationNames.ShadowIndexAnnotation
-        };
-
         public CSharpModelGenerator([NotNull] CSharpHelper code)
         {
             Check.NotNull(code, nameof(code));
@@ -410,7 +404,7 @@ namespace Microsoft.Data.Entity.Commands.Migrations
             Check.NotNull(annotations, nameof(annotations));
             Check.NotNull(stringBuilder, nameof(stringBuilder));
 
-            foreach (var annotation in annotations.Where(annotation => !_ignoredAnnotations.Contains(annotation.Name)))
+            foreach (var annotation in annotations.Where(annotation => !MigrationCodeGenerator.IgnoredAnnotations.Contains(annotation.Name)))
             {
                 stringBuilder.AppendLine();
                 GenerateAnnotation(annotation, stringBuilder);

--- a/src/EntityFramework.Commands/Migrations/MigrationCodeGenerator.cs
+++ b/src/EntityFramework.Commands/Migrations/MigrationCodeGenerator.cs
@@ -3,7 +3,9 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using JetBrains.Annotations;
+using Microsoft.Data.Entity.Infrastructure;
 using Microsoft.Data.Entity.Metadata;
 using Microsoft.Data.Entity.Migrations.Operations;
 
@@ -11,6 +13,12 @@ namespace Microsoft.Data.Entity.Commands.Migrations
 {
     public abstract class MigrationCodeGenerator
     {
+        public static IReadOnlyList<string> IgnoredAnnotations { get; } = new List<string>
+        {
+            CoreAnnotationNames.OriginalValueIndexAnnotation,
+            CoreAnnotationNames.ShadowIndexAnnotation
+        };
+
         public abstract string Language { get; }
 
         public abstract string Generate(
@@ -32,5 +40,76 @@ namespace Microsoft.Data.Entity.Commands.Migrations
             [NotNull] Type contextType,
             [NotNull] string modelSnapshotName,
             [NotNull] IModel model);
+
+        protected virtual IEnumerable<string> GetNamespaces(IEnumerable<MigrationOperation> operations)
+            => GetAnnotationNamespaces(GetAnnotatables(operations));
+
+        private IEnumerable<IAnnotatable> GetAnnotatables(IEnumerable<MigrationOperation> operations)
+        {
+            foreach (var operation in operations)
+            {
+                yield return operation;
+
+                var createTableOperation = operation as CreateTableOperation;
+                if (createTableOperation != null)
+                {
+                    foreach (var column in createTableOperation.Columns)
+                    {
+                        yield return column;
+                    }
+
+                    yield return createTableOperation.PrimaryKey;
+
+                    foreach (var uniqueConstraint in createTableOperation.UniqueConstraints)
+                    {
+                        yield return uniqueConstraint;
+                    }
+
+                    foreach (var foreignKey in createTableOperation.ForeignKeys)
+                    {
+                        yield return foreignKey;
+                    }
+                }
+            }
+        }
+
+        protected virtual IEnumerable<string> GetNamespaces(IModel model)
+            => GetAnnotationNamespaces(GetAnnotatables(model));
+
+        private IEnumerable<IAnnotatable> GetAnnotatables(IModel model)
+        {
+            yield return model;
+
+            foreach (var entityType in model.EntityTypes)
+            {
+                yield return entityType;
+
+                foreach (var property in entityType.GetDeclaredProperties())
+                {
+                    yield return property;
+                }
+
+                foreach (var key in entityType.GetDeclaredKeys())
+                {
+                    yield return key;
+                }
+
+                foreach (var foreignKey in entityType.GetDeclaredForeignKeys())
+                {
+                    yield return foreignKey;
+                }
+
+                foreach (var index in entityType.GetDeclaredIndexes())
+                {
+                    yield return index;
+                }
+            }
+        }
+
+        private IEnumerable<string> GetAnnotationNamespaces(IEnumerable<IAnnotatable> items)
+            => from i in items
+               from a in i.Annotations
+               where a.Value != null && !IgnoredAnnotations.Contains(a.Name)
+               select a.Value.GetType().Namespace;
     }
 }

--- a/src/EntityFramework.Relational/Migrations/Builders/CreateTableBuilder.cs
+++ b/src/EntityFramework.Relational/Migrations/Builders/CreateTableBuilder.cs
@@ -112,7 +112,7 @@ namespace Microsoft.Data.Entity.Migrations.Builders
             return new OperationBuilder<AddUniqueConstraintOperation>(operation);
         }
 
-        public new virtual CreateTableBuilder<TColumns> Annotation([NotNull] string name, [NotNull] string value)
+        public new virtual CreateTableBuilder<TColumns> Annotation([NotNull] string name, [NotNull] object value)
             => (CreateTableBuilder<TColumns>)base.Annotation(name, value);
 
         private string[] Map(LambdaExpression columns) =>

--- a/src/EntityFramework.Relational/Migrations/Builders/OperationBuilder.cs
+++ b/src/EntityFramework.Relational/Migrations/Builders/OperationBuilder.cs
@@ -24,7 +24,7 @@ namespace Microsoft.Data.Entity.Migrations.Builders
 
         public virtual OperationBuilder<TOperation> Annotation(
             [NotNull] string name,
-            [NotNull] string value)
+            [NotNull] object value)
         {
             Check.NotEmpty(name, nameof(name));
             Check.NotNull(value, nameof(value));

--- a/src/EntityFramework.SqlServer/Migrations/SqlServerMigrationAnnotationProvider.cs
+++ b/src/EntityFramework.SqlServer/Migrations/SqlServerMigrationAnnotationProvider.cs
@@ -1,9 +1,7 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
 using System.Collections.Generic;
-using System.Linq;
 using Microsoft.Data.Entity.Infrastructure;
 using Microsoft.Data.Entity.Metadata;
 using Microsoft.Data.Entity.Migrations.Infrastructure;
@@ -14,10 +12,24 @@ namespace Microsoft.Data.Entity.SqlServer.Migrations
     public class SqlServerMigrationAnnotationProvider : MigrationAnnotationProvider
     {
         public override IEnumerable<IAnnotation> For(IKey key)
-            => key.Annotations.Where(a => a.Name == SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.Clustered);
+        {
+            if (key.SqlServer().IsClustered.HasValue)
+            {
+                yield return new Annotation(
+                     SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.Clustered,
+                     key.SqlServer().IsClustered.Value);
+            }
+        }
 
         public override IEnumerable<IAnnotation> For(IIndex index)
-            => index.Annotations.Where(a => a.Name == SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.Clustered);
+        {
+            if (index.SqlServer().IsClustered.HasValue)
+            {
+                yield return new Annotation(
+                     SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.Clustered,
+                     index.SqlServer().IsClustered.Value);
+            }
+        }
 
         public override IEnumerable<IAnnotation> For(IProperty property)
         {

--- a/src/EntityFramework.SqlServer/Migrations/SqlServerMigrationSqlGenerator.cs
+++ b/src/EntityFramework.SqlServer/Migrations/SqlServerMigrationSqlGenerator.cs
@@ -276,10 +276,9 @@ namespace Microsoft.Data.Entity.SqlServer
                 model,
                 builder);
 
-            var valueGeneration = (SqlServerIdentityStrategy?)annotatable[
-                SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.ValueGenerationStrategy];
-
-            if (valueGeneration == SqlServerIdentityStrategy.IdentityColumn)
+            var valueGenerationStrategy = annotatable[
+                SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.ValueGenerationStrategy] as SqlServerIdentityStrategy?;
+            if (valueGenerationStrategy == SqlServerIdentityStrategy.IdentityColumn)
             {
                 builder.Append(" IDENTITY");
             }
@@ -354,10 +353,10 @@ namespace Microsoft.Data.Entity.SqlServer
             Check.NotNull(operation, nameof(operation));
             Check.NotNull(builder, nameof(builder));
 
-            var clustered = (string)operation[SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.Clustered];
-            if (clustered != null)
+            var clustered = operation[SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.Clustered] as bool?;
+            if (clustered.HasValue)
             {
-                builder.Append(clustered == "True" ? "CLUSTERED " : "NONCLUSTERED ");
+                builder.Append(clustered.Value ? "CLUSTERED " : "NONCLUSTERED ");
             }
         }
 

--- a/src/EntityFramework.Sqlite/Migrations/SqliteMigrationSqlGenerator.cs
+++ b/src/EntityFramework.Sqlite/Migrations/SqliteMigrationSqlGenerator.cs
@@ -91,23 +91,20 @@ namespace Microsoft.Data.Entity.Sqlite.Migrations
                 schema, table, name, clrType, type, nullable,
                 defaultValue, defaultValueSql, computedColumnSql, annotatable, model, builder);
 
-            var columnAnnotation = annotatable as Annotatable;
-            var inlinePk = columnAnnotation?.FindAnnotation(SqliteAnnotationNames.Prefix + SqliteAnnotationNames.InlinePrimaryKey);
-
-            if (inlinePk != null
-                && (bool)inlinePk.Value)
+            var inlinePk = annotatable[SqliteAnnotationNames.Prefix + SqliteAnnotationNames.InlinePrimaryKey] as bool?;
+            if (inlinePk == true)
             {
-                var inlinePkName = columnAnnotation.FindAnnotation(SqliteAnnotationNames.Prefix + SqliteAnnotationNames.InlinePrimaryKeyName);
-                if (!string.IsNullOrEmpty(inlinePkName?.Value as string))
+                var inlinePkName = annotatable[
+                    SqliteAnnotationNames.Prefix + SqliteAnnotationNames.InlinePrimaryKeyName] as string;
+                if (!string.IsNullOrEmpty(inlinePkName))
                 {
                     builder
                         .Append(" CONSTRAINT ")
-                        .Append(_sql.DelimitIdentifier((string)inlinePkName.Value));
+                        .Append(_sql.DelimitIdentifier(inlinePkName));
                 }
                 builder.Append(" PRIMARY KEY");
-                var autoincrement = columnAnnotation.FindAnnotation(SqliteAnnotationNames.Prefix + SqliteAnnotationNames.Autoincrement);
-                if (autoincrement != null
-                    && (bool)autoincrement.Value)
+                var autoincrement = annotatable[SqliteAnnotationNames.Prefix + SqliteAnnotationNames.Autoincrement] as bool?;
+                if (autoincrement == true)
                 {
                     builder.Append(" AUTOINCREMENT");
                 }

--- a/test/EntityFramework.Commands.FunctionalTests/Migrations/CodeCompilationTest.cs
+++ b/test/EntityFramework.Commands.FunctionalTests/Migrations/CodeCompilationTest.cs
@@ -3,6 +3,7 @@
 
 using System;
 using System.Reflection;
+using System.Text.RegularExpressions;
 using Microsoft.Data.Entity.Commands.TestUtilities;
 using Microsoft.Data.Entity.Commands.Utilities;
 using Microsoft.Data.Entity.Metadata;
@@ -29,13 +30,21 @@ namespace Microsoft.Data.Entity.Commands.Migrations
             var migrationCode = generator.Generate(
                 "MyNamespace",
                 "MyMigration",
-                new MigrationOperation[0],
+                new[] {
+                    new SqlOperation
+                    {
+                        Sql = "-- TEST",
+                        ["Some:EnumValue"] = RegexOptions.Multiline
+                    }
+                },
                 new MigrationOperation[0]);
             Assert.Equal(
-                @"using System.Collections.Generic;
+                @"using System;
+using System.Collections.Generic;
 using Microsoft.Data.Entity.Migrations;
 using Microsoft.Data.Entity.Migrations.Builders;
 using Microsoft.Data.Entity.Migrations.Operations;
+using System.Text.RegularExpressions;
 
 namespace MyNamespace
 {
@@ -43,6 +52,9 @@ namespace MyNamespace
     {
         public override void Up(MigrationBuilder migration)
         {
+            migration.Sql(
+                ""-- TEST"")
+                .Annotation(""Some:EnumValue"", RegexOptions.Multiline);
         }
 
         public override void Down(MigrationBuilder migration)
@@ -59,13 +71,14 @@ namespace MyNamespace
                 "MyMigration",
                 "20150511161616_MyMigration",
                 "7.0.0",
-                new Model());
+                new Model {["Some:EnumValue"] = RegexOptions.Multiline });
             Assert.Equal(
                 @"using System;
 using Microsoft.Data.Entity;
 using Microsoft.Data.Entity.Metadata;
 using Microsoft.Data.Entity.Migrations.Infrastructure;
 using Microsoft.Data.Entity.Commands.Migrations;
+using System.Text.RegularExpressions;
 
 namespace MyNamespace
 {
@@ -84,6 +97,8 @@ namespace MyNamespace
 
         public override void BuildTargetModel(ModelBuilder builder)
         {
+            builder
+                .Annotation(""Some:EnumValue"", RegexOptions.Multiline);
         }
     }
 }
@@ -95,7 +110,10 @@ namespace MyNamespace
                 References =
                 {
                     BuildReference.ByName(typeof(CodeCompilationTest).GetTypeInfo().Assembly.GetName().Name),
-#if !DNXCORE50
+#if DNXCORE50
+                    BuildReference.ByName("System.Text.RegularExpressions"),
+#else
+                    BuildReference.ByName("System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089"),
                     BuildReference.ByName("System.Runtime, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"),
 #endif
                     BuildReference.ByName("EntityFramework.Core"),
@@ -119,7 +137,7 @@ namespace MyNamespace
 
             var migrationBuilder = new MigrationBuilder();
             migration.Up(migrationBuilder);
-            Assert.Empty(migrationBuilder.Operations);
+            Assert.Equal(1, migrationBuilder.Operations.Count);
 
             migrationBuilder = new MigrationBuilder();
             migration.Down(migrationBuilder);
@@ -140,12 +158,17 @@ namespace MyNamespace
                 new CSharpMigrationOperationGenerator(codeHelper),
                 new CSharpModelGenerator(codeHelper));
 
-            var modelSnapshotCode = generator.GenerateSnapshot("MyNamespace", typeof(MyContext), "MySnapshot", new Model());
+            var modelSnapshotCode = generator.GenerateSnapshot(
+                "MyNamespace",
+                typeof(MyContext),
+                "MySnapshot",
+                new Model {["Some:EnumValue"] = RegexOptions.Multiline });
             Assert.Equal(@"using System;
 using Microsoft.Data.Entity;
 using Microsoft.Data.Entity.Metadata;
 using Microsoft.Data.Entity.Migrations.Infrastructure;
 using Microsoft.Data.Entity.Commands.Migrations;
+using System.Text.RegularExpressions;
 
 namespace MyNamespace
 {
@@ -154,6 +177,8 @@ namespace MyNamespace
     {
         public override void BuildModel(ModelBuilder builder)
         {
+            builder
+                .Annotation(""Some:EnumValue"", RegexOptions.Multiline);
         }
     }
 }
@@ -164,7 +189,10 @@ namespace MyNamespace
                 References =
                 {
                     BuildReference.ByName(typeof(CodeCompilationTest).GetTypeInfo().Assembly.GetName().Name),
-#if !DNXCORE50
+#if DNXCORE50
+                    BuildReference.ByName("System.Text.RegularExpressions"),
+#else
+                    BuildReference.ByName("System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089"),
                     BuildReference.ByName("System.Runtime, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"),
 #endif
                     BuildReference.ByName("EntityFramework.Core"),

--- a/test/EntityFramework.SqlServer.Tests/Migrations/SqlServerMigrationSqlGeneratorTest.cs
+++ b/test/EntityFramework.SqlServer.Tests/Migrations/SqlServerMigrationSqlGeneratorTest.cs
@@ -81,7 +81,7 @@ namespace Microsoft.Data.Entity.SqlServer.Migrations
                 {
                     Table = "People",
                     Columns = new[] { "Id" },
-                    [SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.Clustered] = bool.FalseString
+                    [SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.Clustered] = false
                 });
 
             Assert.Equal(
@@ -129,7 +129,7 @@ namespace Microsoft.Data.Entity.SqlServer.Migrations
                     Name = "IX_People_Name",
                     Table = "People",
                     Columns = new[] { "Name" },
-                    [SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.Clustered] = bool.TrueString
+                    [SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.Clustered] = true
                 });
 
             Assert.Equal(

